### PR TITLE
OOP school library: decorate a class

### PR DESCRIPTION
**Pull request by Laylo309**
- [x]  Create a class `Corrector` with a method `correct_name` that does the following:
  - Make sure the first letter of the given name is a capital letter.
  - Make sure that the name has a maximum of 10 characters. If it's longer it should trim the word.

- [x]  Set an instance of `Corrector` in `Person` on initialization.

- [x] In `Person` add a method `validate_name` that uses the `Corrector` instance to validate and save the corrected name.

### CODE REVIEW

File: corrector.rb [hunk 1]
```diff
@@ -0,0 +1,8 @@
+require_relative './person'
+
+class Corrector
+  def correct_name(name: 'Unknown')
+    name.capitalize
```
Comment (o-rashad) [inline corrector.rb]: Great work on creating the `Corrector` class with `correct_name` method... 💯 
only one thing i want to mention is 

 - `capitalize` returns a new string with the first letter capitalized
 - `capitalize!` modifies or mutates the original string 

For further read [LINK](https://stackoverflow.com/questions/26103869/using-capitalize-or-capitalize)


- [ ] Change `capitalize` to `capitalize!` to mutate the user input name to capitalize it if it's not. 👍

File: person.rb [hunk 2]
```diff
@@ -7,12 +8,17 @@ def initialize(age:, name: 'Unknown', parent_permission: true)
     @name = name
     @age = age
     @parent_permission = parent_permission
+    @corrector = corrector
```
Comment (o-rashad) [inline person.rb]: To be able to create an instance of a any class we need to use the `new()` method 👍 

For example : ⬇️ 
 - `@animal= Animal.new()`
 
 More info [LINK](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/articles/composition_by_examples.md#composable-classes)
 
 
 - [ ] Change `@corrector` instance variable to

File: person.rb [hunk 3]
```diff
@@ -7,12 +8,17 @@ def initialize(age:, name: 'Unknown', parent_permission: true)
...
   end
 
   def can_use_services?
     of_age? || true
   end
 
+  def validate_name
+    @name = @corrector.correct_name(name)
```
Comment (o-rashad) [inline person.rb]: Nicely done, you used the Corrector instance to validate and save the corrected name. 😎👍

**Review by o-rashad — CHANGES_REQUESTED** [review]
# Status: Changes Required ♻️

Asslam Alikom @Laylo309 👋😄,
I'm Omar and I will be your reviewer for this project 😊

Good job so far!👌😃

There are some issues 🐞 that you still need to work on to go to the next project but you are almost there! 🏁🏁🏁

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: person.rb (hunk 3 above)
Comment (Laylo309) [reply person.rb]: Done ✔

File: person.rb (hunk 2 above)
Comment (Laylo309) [reply person.rb]: Done ✔

File: corrector.rb (hunk 1 above)
Comment (Laylo309) [reply corrector.rb]: Done ✔

**Review by mikethreels — APPROVED** [review]
Hi Laylo, 

Your project is complete!  There is nothing else to say other than... it's time to merge it 🚀 :
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**